crypto/uid.c

```diff
@@ -31,12 +31,18 @@ int OPENSSL_issetugid(void)
 # include OPENSSL_UNISTD
 # include <sys/types.h>
 
+# if defined(__GLIBC__) && defined(__GLIBC_PREREQ)
+#  if __GLIBC_PREREQ(2, 16)
+#   include <sys/auxv.h>
+#  endif
+# endif
+
 int OPENSSL_issetugid(void)
 {
-    if (getuid() != geteuid())
-        return 1;
-    if (getgid() != getegid())
-        return 1;
-    return 0;
+# ifdef AT_SECURE
+    return getauxval(AT_SECURE);
```

Comment:
Should this be != 0? It's documented to return a non-zero value.

Reply:
issetugid() is documented to return 1. This function seems to be undocumented.

Reply:
I guess it doesn't really matter

Reply:
I've added the != 0.

Reply:
It is documented in recent glibc's.  It returns an unsigned long and in this case a non-zero value means privileged.  The not zero check is appropriate and necessary.